### PR TITLE
docs: update RTE value docs to reflect default format change (v24)

### PR DESCRIPTION
**Pull request by sissbruecker**
Updates the RichTextEditor "Value Format" section to reflect changes in v24 where the HTML format becomes the default for the Flow component.

Also updates the code examples to demonstrate usage of both formats.

This should only be merged after v24 has been released and docs have been upgraded.

### CODE REVIEW

File: src/main/java/com/vaadin/demo/component/richtexteditor/RichTextEditorSetGetValue.java [hunk 1]
```diff
@@ -11,21 +11,32 @@ public class RichTextEditorSetGetValue extends Div {
...
+                "Enter something in the Rich Text Editor to see its Delta value here.");
+        deltaTextArea.setWidthFull();
+        rte.asDelta().addValueChangeListener(e -> deltaTextArea.setValue(e.getValue()));
+        deltaTextArea.addValueChangeListener(e -> {
+            if (!rte.asDelta().getValue().equals(e.getValue())) {
+                rte.asDelta().setValue(e.getValue());
+            }
+        });
+
+        add(rte, htmlTextArea);
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/richtexteditor/RichTextEditorSetGetValue.java:39]: `deltaTextArea` should also be added

File: frontend/demo/component/richtexteditor/rich-text-editor-set-get-value.ts [hunk 2]
```diff
@@ -28,23 +30,39 @@ export class Example extends LitElement {
...
+        @change="${(e: TextAreaChangeEvent) => this.setHtmlValue(e.target.value)}"
+      ></vaadin-text-area>
+
+      <vaadin-text-area
+        label="Delta Value"
+        placeholder="Enter something in the Rich Text Editor to see its Delta value here."
+        style="width: 100%;"
+        .value="${this.deltaValue}"
+        @change="${(e: TextAreaChangeEvent) => {
+          this.deltaValue = e.target.value;
```
Comment (tomivirkki) [inline frontend/demo/component/richtexteditor/rich-text-editor-set-get-value.ts:54]: Updating one of the format value fields doesn't affect the other one, only the RTE:

https://user-images.githubusercontent.com/1222264/218084206-e80d2382-230b-44fb-8aea-5dcf31f3775d.mp4

File: src/main/java/com/vaadin/demo/component/richtexteditor/RichTextEditorSetGetValue.java (hunk 1 above)
Comment (sissbruecker) [reply src/main/java/com/vaadin/demo/component/richtexteditor/RichTextEditorSetGetValue.java]: Thanks, fixed.

File: frontend/demo/component/richtexteditor/rich-text-editor-set-get-value.ts (hunk 2 above)
Comment (sissbruecker) [reply frontend/demo/component/richtexteditor/rich-text-editor-set-get-value.ts:61]: I think I ran into an infinite loop when trying this earlier. Trying it now it seems to work, so it's fixed.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**